base/process.jl

```diff
@@ -72,6 +72,10 @@ backticks, e.g.
 """
 Cmd
 
+hash(x::Cmd) = hash(x.exec, hash(x.env, hash(x.ignorestatus, hash(x.dir, hash(x.flags)))))
+==(x::Cmd, y::Cmd) = isequal(x.exec, y.exec) && isequal(x.env, y.env) &&
```

Comment:
As @stevengj noted, you should use `==` instead of `isequal`. The latter only makes a difference for numeric types anyway.